packages/peregrine/lib/talons/CartPage/useCartPage.js

```diff
@@ -36,9 +36,24 @@ export const useCartPage = props => {
     const hasItems = !!(data && data.cart.total_quantity);
     const shouldShowLoadingIndicator = called && loading && !hasItems;
 
+    const hasOutOfStockItem = useMemo(() => {
```

Comment:
I realize this is duplicated three times and we should probably just move to a `StockStatusMessage` component. Waiting on UX to finalize design, and will re-factor this duplication out before sending to code review.